service/type.go

```diff
@@ -0,0 +1,97 @@
+package service
+
+import "github.com/mesg-foundation/engine/hash"
+
+// MainServiceKey is key for main service.
+const MainServiceKey = "service"
+
+// StatusType of the service.
+type StatusType uint
+
+// Possible statuses for service.
+const (
+	UNKNOWN StatusType = iota
+	STOPPED
+	STARTING
+	PARTIAL
+	RUNNING
+)
+
+func (s StatusType) String() string {
+	switch s {
+	case STOPPED:
+		return "STOPPED"
+	case STARTING:
+		return "STARTING"
+	case PARTIAL:
+		return "PARTIAL"
+	case RUNNING:
+		return "RUNNING"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// WARNING about hash tags on Service type and its inner types:
+// * never change the name attr of hash tag. use an incremented value for
+// name attr when a new configuration field added to Service.
+// * don't increment the value of name attr if corresponding field's name
+// changed but its behavior remains the same.
+// * this is required for not breaking Service IDs unless there is a behavioral
+// change.
+
+// Service represents a MESG service.
+type Service struct {
+	Hash          hash.Hash     `hash:"-" validate:"required"`                                // Hash is calculated from the combination of service's source and mesg.yml. It represents the service uniquely.
```

Comment:
I recommend to keep the comment on top of the parameter. Nice to read and more space to put more comments.

```suggestion
	// Hash is calculated from the combination of service's source and mesg.yml. It represents the service uniquely.
	Hash          hash.Hash     `hash:"-" validate:"required"`
```

Reply:
I knew this was too much, but it looked nice